lib/vtls/unitytls.c

```diff
@@ -220,18 +220,78 @@ static size_t on_read(void* userData, UInt8* buffer, size_t bufferLen, unitytls_
   return read;
 }
 
+/* Pretty much a copy of Curl_send_plain */
```

Comment:
Difference being that the other one takes a Curl_easy, right? Can we create one on the fly for this purpose so we don't need to duplicate the code?

Reply:
I can see from your commit history that you struggled with using a curl_easy directly in the connectdata. This sounds familiar to me, I believe I tried the same, but can't recall why it didn't work out. Maybe we should slack a bit on this

Reply:
For the record: Curl_easy contains a lot of stuff (all option etc.) and the only reason Curl_send_plain uses Curl_easy is for storing error message string. Originally connection struct had a pointer to Curl_easy in it, but now it has a list of them and we don't know which one is the correct one (would have to modify the context and catch connection re-use somehow). So this looks to be a simplest approach, though with a risk that original curl code might get updated and ours not getting improvements.